core/types/uuid.go

```diff
@@ -62,6 +87,134 @@ func (u *UUID) Scan(src any) error {
 	return errors.New("not a byte slice")
 }
 
+func (u *UUID) FormatSupported(format int16) bool {
+	return format == pgtype.TextFormatCode || format == pgtype.BinaryFormatCode
+}
+
+func (u *UUID) PreferredFormat() int16 {
+	return pgtype.TextFormatCode
+}
+
+func (u *UUID) PlanEncode(m *pgtype.Map, oid uint32, format int16, value any) pgtype.EncodePlan {
+	if _, ok := value.(*UUID); !ok {
+		return nil // not sure why we do this, but pgx does it
+	}
+
+	switch format {
+	// given our uuid type, I believe it will always come binary
+	case pgtype.BinaryFormatCode:
+		return encodePlanFunc(func(value any, buf []byte) (newBuf []byte, err error) {
+			uuid, ok := value.(*UUID)
+			if !ok {
+				return nil, errors.New("internal error: expected UUID")
+			}
+
+			return append(buf, uuid[:]...), nil
+		})
+	case pgtype.TextFormatCode:
+		return encodePlanFunc(func(value any, buf []byte) (newBuf []byte, err error) {
+			uuid, ok := value.(*UUID)
+			if !ok {
+				return nil, errors.New("internal error: expected *UUID")
+			}
+
+			return append(buf, uuid.String()...), nil
+		})
+	}
+
+	return nil
+}
+
+type encodePlanFunc func(value any, buf []byte) (newBuf []byte, err error)
+
+func (e encodePlanFunc) Encode(value any, buf []byte) (newBuf []byte, err error) {
+	return e(value, buf)
+}
+
+func (u *UUID) PlanScan(m *pgtype.Map, oid uint32, format int16, target any) pgtype.ScanPlan {
+	switch format {
+	case pgtype.BinaryFormatCode:
+		return scanPlanFunc(func(src []byte, target any) error {
+			if target == nil {
+				return nil
+			}
+
+			if len(src) == 0 {
+				return nil
+			}
+
+			uuid, ok := target.(*UUID)
+			if !ok {
+				if len(src) != 16 {
+					return fmt.Errorf("expected 16 bytes, got %d", len(src))
+				}
+
+				uuid := UUID{}
+				copy(uuid[:], src)
+				reflect.ValueOf(target).Elem().Set(reflect.ValueOf(uuid))
```

Comment:
What's this doing?  I'm the case where `!ok` above, what's an example with concrete types where this does something sensible?
I'm not sure where `target` comes from or is used later

Reply:
I'll document this. When testing, it sometimes did not pass a UUID, but instead passed []byte. I believe this is because of how we pass uuids internally (such as in the vote store), where we convert them to []byte.